### PR TITLE
Add dynamic shape support for cumsum/grid

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Add dynamic shape support for cumsum/grid

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by peri044 — APPROVED** [review]
LGTM, pending CI.

File: py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py [hunk 1]
```diff
@@ -387,17 +387,46 @@ def cumsum(
...
+    if input_shape[dim] < 0:
+        trip_limit = impl.shape.shape(
+            ctx, target, source_ir, name + "_shape", input, dim
+        )
+        # the trip_limit has to be a 0D shape tensor, however this impl.shape.shape gives a 1D shape
+        # for example if the trip limit is 3, it wants a tensor(3), not a tensor([3])
+        # in order to reduce it from 1D to 0D, i have to use this impl.reduce.sum
+        trip_limit = impl.reduce.sum(
+            ctx, target, source_ir, name, trip_limit, 0, keepdim=False
+        )
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py:402]: Another way to create a 0D shape tensor you can refer to: https://github.com/pytorch/TensorRT/blob/main/py/torch_tensorrt/dynamo/conversion/impl/embedding.py#L191

File: py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py [hunk 2]
```diff
@@ -387,17 +387,46 @@ def cumsum(
...
-    zeros = np.zeros(new_dims)
-    zero_trttensor = get_trt_tensor(ctx, zeros, f"{name}_initial_value")
+    if has_dynamic_shape(data.shape):
+        data_shape = []
+        for i in range(len(input_shape)):
+            if i != dim:
+                if input_shape[i] < 0:
+                    data_shape.append(
+                        impl.shape.shape(
+                            ctx, target, source_ir, name + f"_{i=}_shape", input, i
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/slice/ops.py:418]: Is name here correct?

**Review by zewenli98 — COMMENTED** [review]
LGTM, just a few minor comments.

**Pull request merged**